up-l3/udiscovery/v3/service.adoc

```diff
@@ -0,0 +1,250 @@
+
+----
+SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+
+See the NOTICE file(s) distributed with this work for additional
+information regarding copyright ownership.
+
+This program and the accompanying materials are made available under
+the terms of the Apache License Version 2.0 which is available at
+https://www.apache.org/licenses/LICENSE-2.0
+ 
+SPDX-FileType: DOCUMENTATION
+SPDX-License-Identifier: Apache-2.0
+----
+
+== Overview
+
+UDiscovery provides the means for uEs to discover services (and their metadata), for local and remote services. Implementation of uDiscovery takes inspiration from Domain Name System (DNS) principles in that we store and cache queries and replicate local information to more centralized databases. The local uDiscovery service acts as the first point of contact for discovery and then the discovery service looks up at the domain level service for information about services across a domain (i.e. vehicle). The central instance of uDiscovery service  is the last point of contact for global searches of services.
+
+UDiscovery Service implements two sets of APIs:
+
+- *Client APIs (link:../../../up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto[udiscovery.proto]):* Used by uEs to find services and their topic metadata, and for recursive lookups at the domain and central levels.
+
+- *Replication API (link:../../../up-core-api/uprotocol/core/udiscovery/v3/udiscovery_replicator.proto[udiscovery_replicator.proto]):* Used Internally by uDiscovery services to add, update, or delete service metadata from local, domain, and/or central service instances.
+
+<<udiscovery-design>> below provides an overview of the design and use of the Client and Replication APIs to implement the uDiscovery business logic.
+
+.uProtocol Discovery Design
+[#udiscovery-design]
+image::design.drawio.svg[Overview]
+
+=== Terms
+
+- *Fully Qualified Service URI (FQURI):* A URI that contains all the information needed to uniquely identify a service. A fully qualified service URI contains the following information:
+  - *ue_id:* The unique identifier of the service (both the instance and service ID portions).
+  - *authority_name:* The name of the authority that hosts the service.
+  - *ue_version_major:* The version of the service.
+  - *resource_id:* The resource ID (value of 0 us used to indicate the default ID for the service).
+
+
+== Client APIs
+
+The following section we will elaborate on the service-side requirements for the uDiscover service when it serves the client facing APIs defined in the xref:../../../up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto[udiscovery.proto].
+
+=== FindServices()
+
+`FindServices()` is used to find the authority(ies), instance(s), and version(s) of the serice id (ue_id). The API is passed `FindServicesRequest` message and it returns a `FindServicesResponse` message that contains a batch of 1 or more services that match the search criteria.
```

Comment:
IMHO we can omit the specific names of the request and response messages (defined in the proto anyway) and focus on their semantics.